config/config_file.go

```diff
@@ -52,9 +52,21 @@ type ConfigFile struct {
 }
 
 type AuthConfig struct {
-	Gateway string   `yaml:"gateway,omitempty"`
-	Auth    AuthType `yaml:"auth,omitempty"`
-	Token   string   `yaml:"token,omitempty"`
+	Gateway      string       `yaml:"gateway,omitempty"`
+	Auth         AuthType     `yaml:"auth,omitempty"`
+	Token        string       `yaml:"token,omitempty"`
+	Oauth2Config Oauth2Config `yaml:"oauth,omitempty"`
+}
+
+type Oauth2Config struct {
```

Comment:
I think this would work better as a KV - similar to how env-vars are represented in K8s YAML.

Reply:
This has been converted to use a KV format.